OrbitQt/servicedeploymanager.cpp

```diff
@@ -95,7 +96,13 @@ outcome::result<uint16_t> ServiceDeployManager::StartTunnel(
 
   tunnel->value().Start();
 
-  OUTCOME_TRY(loop_.exec());
+  outcome::result<int> tunnel_result = loop_.exec();
+  if (!tunnel_result) {
+    ERROR("Unable to start tunnel, error message: %s",
+          tunnel_result.error().message().c_str());
+    return Error::kCouldNotStartTunnel;
+  }
+
```

Comment:
Could you wrap this pattern into a free function.
It would be nice if we could keep the OUTCOME_TRY here. Something like this:
`OUTCOME_TRY(mapError(loop_.exec(), Error::kCouldNotStartTunnel));`

Reply:
I wasn't sure If I had expressed myself clearly, so I wrote an example:

```cpp
template <typename T>
static outcome::result<T> MapError(outcome::result<T> result, Error new_error) {
  if (result) {
    return result;
  } else {
    const auto new_error_code = make_error_code(new_error);
    ERROR("%s: %s", new_error_code->message(), result.error().message();
    return outcome::failure(new_error_code);
  }
}
```

Reply:
Thanks for taking the time to do the example. Makes my integration very easy!

Reply:
done